planner/core/rule_join_elimination.go

```diff
@@ -73,33 +84,20 @@ func (o *outerJoinEliminator) extractInnerJoinKeys(join *LogicalJoin, innerChild
 	return expression.NewSchema(joinKeys...)
 }
 
-func (o *outerJoinEliminator) isAggColsAllFromOuterTable(outerPlan LogicalPlan, aggCols []*expression.Column) (bool, error) {
-	if len(aggCols) == 0 {
-		return false, nil
-	}
-	for _, col := range aggCols {
-		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
-		c, err := outerPlan.Schema().FindColumn(columnName)
-		if err != nil || c == nil {
-			return false, err
-		}
-	}
-	return true, nil
-}
-
-// check whether schema cols of join's parent plan are all from outer join table
-func (o *outerJoinEliminator) isParentColsAllFromOuterTable(outerPlan LogicalPlan, parentSchema *expression.Schema) (bool, error) {
-	if parentSchema == nil {
-		return false, nil
-	}
-	for _, col := range parentSchema.Columns {
-		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
-		c, err := outerPlan.Schema().FindColumn(columnName)
-		if err != nil || c == nil {
-			return false, err
+// check whether the cols all from outer plan
+func (o *outerJoinEliminator) isColsAllFromOuterTable(outerPlan LogicalPlan, cols []*expression.Column, outerUniqueIDs set.Int64Set) bool {
+	// There are two cases "return false" here:
+	// 1. If cols represents aggCols, then "len(cols) == 0" means no duplicate agnostic aggregate functions before.
```

Comment:
Should be `not all aggregate functions are duplicate agnostic`?

Reply:
Changed.